include/soci/connection-parameters.h

```diff
@@ -32,26 +32,33 @@ class SOCI_DECL connection_parameters
 
     // Retrieve the backend and the connection strings specified in the ctor.
     backend_factory const * get_factory() const { return factory_; }
-    void set_connect_string(const std::string & connectString) { connectString_ = connectString; }
+    void set_connect_string(std::string const & connectString)
+    {
+        connectString_ = connectString;
+        parse();
+    }
     std::string const & get_connect_string() const { return connectString_; }
 
     // Set the value of the given option, overwriting any previous value.
-    void set_option(const char * name, std::string const & value)
+    void set_option(std::string const & name, std::string const & value)
```

Comment:
I'm not sure I understand it. Regardless if `value` below is `char*` or `string const&`, copy is made:
```
map<string, string> m;
m[key] = value;
```